pages/categoryPage/js/category-page-script.js

```diff
@@ -0,0 +1,228 @@
+        const minVal = parseInt(rangeMinInput.value);
+        const maxVal = parseInt(rangeMaxInput.value);
+
+        if (maxVal - minVal < priceGap) {
+            rangeMinInput.value = maxVal - priceGap;
+        }
+
+        updateProgress();
+        updateRangeInputs();
+    });
+
+    rangeMaxInput.addEventListener('input', () => {
+        const minVal = parseInt(rangeMinInput.value);
+        const maxVal = parseInt(rangeMaxInput.value);
+
+        if (maxVal - minVal < priceGap) {
+            rangeMaxInput.value = minVal + priceGap;
+        }
+
+        updateProgress();
+        updateRangeInputs();
+    });
+
+    inputMinPrice.addEventListener('input', () => {
+        let minVal = parseInt(inputMinPrice.value);
+        let maxVal = parseInt(inputMaxPrice.value);
+
+        if (minVal < 0 || minVal > maxPrice-priceGap) {
+            minVal = minPrice;
+            inputMinPrice.value = minPrice;
+        }
+
+        if ((maxVal - minVal >= priceGap)) {
+            rangeMinInput.value = minVal;
+        }
+
+        updateProgress();
+        updatePriceInputs();
+    });
+
+    inputMaxPrice.addEventListener('input', () => {
+        let minVal = parseInt(inputMinPrice.value);
+        let maxVal = parseInt(inputMaxPrice.value);
+
+        if (maxVal > maxPrice) {
+            maxVal = maxPrice;
+            inputMaxPrice.value = maxPrice;
+        }
+
+        if ((maxVal - minVal >= priceGap) && maxVal <= maxPrice) {
+            rangeMaxInput.value = maxVal;
+        }
+
+        updateProgress();
+        updatePriceInputs();
+    });
+
+    rangeMinInput.min = minPrice;
+    rangeMinInput.max = maxPrice;
+    rangeMinInput.value = minPrice;
+
+    rangeMaxInput.min = minPrice;
+    rangeMaxInput.max = maxPrice;
+    rangeMaxInput.value = maxPrice;
+
+    inputMinPrice.value = minPrice;
+    inputMaxPrice.value = maxPrice;
+
+    updateProgress();
+    updateRangeInputs();
+}
+
+function setPriceValuesForPriceFilter(products) {
+    let minPrice = Number.MAX_VALUE;
+    let maxPrice = Number.MIN_VALUE;
+
+    products.forEach(product => {
+        const price = parseFloat(product.variants[0].price);
+        if (price < minPrice) {
+            minPrice = price;
+        }
+        if (price > maxPrice) {
+            maxPrice = price;
+        }
+    });
+
+    priceSliderLogic(minPrice, maxPrice);
+}
+
+async function getProducts() {
+    return await fetch('http://localhost:3000/products')
+        .then(response => response.json())
+        .then(products => {
+            dynamicProductsLoading(products);
+            return products;
+        })
+        .catch(error => {
+            console.error('Failed to fetch products:', error);
+        });
+}
+
+function dynamicProductsLoading(products) {
+    const productContainer = document.querySelector('.product-cards-container');
+
+    products.forEach(product => {
+        const productCard = document.createElement('div');
```

Comment:
Try to use 'template string'
For example:

```js
 `<div class="product-card">
<img src="${product.image}" alt="${product.name}">
...
</div>
`
```

Reply:
Thanks for the example, now I understand what you meant.
Already fixed it.

Reply:
Great work